ssm_jax/hmm/models/mvn_tied_hmm.py

```diff
@@ -0,0 +1,188 @@
+            sum_xxT=(self.num_obs, self.num_obs),
+        )
+
+    def log_prior(self):
+        lp = tfd.Dirichlet(self._initial_probs_concentration.value).log_prob(self.initial_probs.value)
+        lp += tfd.Dirichlet(self._transition_matrix_concentration.value).log_prob(self.transition_matrix.value).sum()
+
+        lp += NormalInverseWishart(self._emission_prior_mean.value, self._emission_prior_conc.value,
+                                   self._emission_prior_df.value, self._emission_prior_scale.value).log_prob(
+                                       (self.emission_covariance_matrices.value, self.emission_means.value)).sum()
+        return lp
+
+    def _zeros_like_suff_stats(self):
+        dim = self.num_obs
+        num_states = self.num_states
+        return MultivariateNormalTiedHMMSuffStats(
+            marginal_loglik=0.0,
+            initial_probs=jnp.zeros((num_states,)),
+            trans_probs=jnp.zeros((num_states, num_states)),
+            sum_w=jnp.zeros((num_states,)),
+            sum_x=jnp.zeros((num_states, dim)),
+            sum_xxT=jnp.zeros((num_states, dim, dim)),
+        )
+
+    # Expectation-maximization (EM) code
+    def e_step(self, batch_emissions):
+        """The E-step computes expected sufficient statistics under the
+        posterior. In the Gaussian case, this these are the first two
+        moments of the data
+        """
+
+        def _single_e_step(emissions):
+            # Run the smoother
+            posterior = hmm_smoother(self._compute_initial_probs(), self._compute_transition_matrices(),
+                                     self._compute_conditional_logliks(emissions))
+
+            # Compute the initial state and transition probabilities
+            initial_probs = posterior.smoothed_probs[0]
+            trans_probs = compute_transition_probs(self.transition_matrix.value, posterior)
+
+            # Compute the expected sufficient statistics
+            sum_w = jnp.einsum("tk->k", posterior.smoothed_probs)
+            sum_x = jnp.einsum("tk,ti->ki", posterior.smoothed_probs, emissions)
+            sum_xxT = jnp.einsum("tk,ti,tj->kij", posterior.smoothed_probs, emissions, emissions)
+
+            # TODO: might need to normalize x_sum and xxT_sum for numerical stability
+            stats = MultivariateNormalTiedHMMSuffStats(marginal_loglik=posterior.marginal_loglik,
+                                                       initial_probs=initial_probs,
+                                                       trans_probs=trans_probs,
+                                                       sum_w=sum_w,
+                                                       sum_x=sum_x,
+                                                       sum_xxT=sum_xxT)
+            return stats
+
+        # Map the E step calculations over batches
+        return vmap(_single_e_step)(batch_emissions)
+
+    def _m_step_emissions(self, batch_emissions, batch_posteriors, **kwargs):
```

Comment:
I think there are some bugs in this math. When the covariance matrix is shared we'll need to have a slightly different M-step. I will explain in a comment in this issue.